Integrations/integration-ArcherRSA.yml

```diff
@@ -0,0 +1,626 @@
+        '               <ShowStatSummaries>false</ShowStatSummaries>'+
+        '               <DisplayFields>'+
+            displayFields +
+            '</DisplayFields>'+
+        '    <Criteria>'+
+        '        <ModuleCriteria>'+
+        '          <Module>' + appId + '</Module>'+
+        '                  </ModuleCriteria>'+
+        '               </Criteria>'+
+        '            </SearchReport> ]]>'+
+        '         </searchOptions>'+
+        '         <pageNumber>1</pageNumber>'+
+        '      </ExecuteSearch>'+
+        '   </soap:Body>'+
+        '</soap:Envelope>';
+    }
+
+    function createContantSoapRequest(token, appId, fields) {
+        return '<?xml version="1.0" encoding="utf-8"?>' +
+           '<soap:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema- instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">' +
+           '     <soap:Body>' +
+           '        <CreateRecord xmlns="http://archer-tech.com/webservices/">' +
+           '             <sessionToken>' + token + '</sessionToken>' +
+           '             <moduleId>' + appId + '</moduleId>' +
+           '             <fieldValues> ' +
+           '                 <![CDATA[ ' +
+           '                    <Record>' +
+                                   fields +
+           '                    </Record>' +
+           '                ]]>' +
+           '            </fieldValues>' +
+           '      </CreateRecord>' +
+           '   </soap:Body> ' +
+           '</soap:Envelope>'
+    }
+
+    function updateContentSoapRequest(token, appId, contentId, fields) {
+        return '<?xml version="1.0" encoding="utf-8"?> ' +
+        '<soap:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchemainstance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">' +
+        ' <soap:Body>' +
+        ' <UpdateRecord xmlns="http://archer-tech.com/webservices/">' +
+        ' <sessionToken>' + token + '</sessionToken>' +
+        ' <moduleId>' + appId + '</moduleId>' +
+        ' <contentId>' + contentId +'</contentId>' +
+        ' <fieldValues>' +
+        '<![CDATA[' +
+            '<UpdateRecord>' +
+                fields +
+            '</UpdateRecord>' +
+        ']]>' +
+        '</fieldValues>' +
+        ' </UpdateRecord>' +
+        ' </soap:Body>' +
+        ' </soap:Envelope>'
+    }
+
+    function getLevelsByApplicationId(token) {
+
+        var lvlResponse = sendRestRequest('levelIdByModuleId', '', token, true);
+        var levels = dq(lvlResponse, 'RequestedObject.Id');
+        console.log('---Levels: ' + levels);
+        return levels;
+    }
+
+    function createMappingFieldIdToName(levels, token) {
+        console.log("in createMappingFieldIdToName");
+        var mapping = {};
+        if(!levels) {
+            throw 'Mapping levels to Fields failed, levels = ' + JSON.stringify(levels);
```

Comment:
level does not exits (can be null) and you are doing JSON.stringify on it.. should be instead:
`throw 'Mapping levels to Fields failed, levels not provided`